server/controllers/compaignRouter/getCampaignById.ts

```diff
@@ -1,16 +1,68 @@
 import { Request, Response, NextFunction } from 'express';
-import { Campaign } from '../../database/models';
+import {
+  Campaign,
+  Capon,
+  Category,
+  Donation,
+  sequelize,
+} from '../../database/models';
 import { CustomedError } from '../../utils';
 import { paramsSchema } from '../../utils/validation';
 
-const getCampaignById = async (req: Request, res: Response, next: NextFunction) => {
+const getCampaignById = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
   try {
     const { id } = await paramsSchema.validateAsync(req.params);
-    const campaignInfo = await Campaign.findByPk(id);
+    const campaignInfo = await Campaign.findByPk(id, {
+      attributes: [
+        'id',
+        'description',
+        'image_link',
+        'is_available',
+        'target',
+        'title',
+      ],
+      include: {
+        model: Category,
+        attributes: ['name', 'icon_url'],
+      },
+    });
+    const current: any = await Donation.findAll({
+      where: {
+        campaignId: id,
+      },
+      raw: true,
+      attributes: [
+        [
+          sequelize.fn(
+            'SUM',
+            sequelize.literal(
+              'COALESCE(food, 0) + COALESCE(clothes, 0) + COALESCE(money, 0)',
+            ),
+          ),
+          'current',
+        ],
+      ],
+    });
+
+    const numOfFamilies = await Capon.findAll({
+      where: {
+        campaignId: id,
+      },
+      raw: true,
+    });
```

Comment:
As all these don't depend on each other you can use PromiseAll.
And you here adding the clothes, money, and food together? they are all different scales.